dbms/src/Core/SpillHandler.cpp

```diff
@@ -63,24 +85,40 @@ void SpillHandler::spillBlocks(const Blocks & blocks)
     try
     {
         Stopwatch watch;
-        RUNTIME_CHECK_MSG(spiller->spill_finished == false, "{}: spill after the spiller is finished.", spiller->config.spill_id);
+        RUNTIME_CHECK_MSG(spiller->isSpillFinished() == false, "{}: spill after the spiller is finished.", spiller->config.spill_id);
         auto block_size = blocks.size();
         LOG_INFO(spiller->logger, "Spilling {} blocks data into temporary file {}", block_size, current_spill_file_name);
+
+        FAIL_POINT_TRIGGER_EXCEPTION(FailPoints::exception_during_spill);
+
         size_t total_rows = 0;
-        if (unlikely(writer == nullptr))
-        {
-            writer = std::make_unique<SpillWriter>(spiller->config.file_provider, current_spill_file_name, blocks[0].cloneEmpty(), spiller->spill_version);
-        }
+        size_t rows_in_file = 0;
+        size_t bytes_in_file = 0;
         for (const auto & block : blocks)
         {
-            total_rows += block.rows();
+            if (unlikely(writer == nullptr))
+            {
+                writer = std::make_unique<SpillWriter>(spiller->config.file_provider, current_spill_file_name, current_append_write, blocks[0].cloneEmpty(), spiller->spill_version);
```

Comment:
How about calling `setUpNextSpilledFile` here, and does not call it in the constructor and L114.
And then the new spill file is not created when no block is written to.

Reply:
Good idea, update the code.